server/src/main/java/com/paypal/selion/grid/servlets/GridForceRestartDelegateServlet.java

```diff
@@ -107,55 +114,26 @@ protected void process(HttpServletRequest request, HttpServletResponse response)
-                    + GridForceRestartDelegateServlet.class.getSimpleName() + "' >");
-            writer.write("<div class='form_description'>");
-            writer.write("<h2>SeLion Grid - Node Restart</h2>");
-            writer.write("<p>Use this page to restart nodes</p>");
-            writer.write("</div>");
-            writer.write("<ul>");
-            writer.write("<li id='li_1' >");
-            writer.write("<label class='description' for='element_1'>List of nodes to restart </label>");
-
-            ProxySet proxies = this.getRegistry().getAllProxies();
-            Iterator<RemoteProxy> iterator = proxies.iterator();
-            boolean nodesPresent = false;
-            while (iterator.hasNext()) {
-                RemoteProxy proxy = iterator.next();
-                if (proxy == null) {
-                    continue;
-                }
-
-                if ((proxy instanceof SeLionRemoteProxy) && (((SeLionRemoteProxy) proxy).supportsForceShutdown())) {
-                    writer.write("<input name='nodes' class='element checkbox' type='checkbox' value='" + proxy.getId()
-                            + "' />");
-                    writer.write(((SeLionRemoteProxy) proxy).isScheduledForRecycle() ? "<label class='choice'>"
-                            + proxy.getId() + " (restart scheduled)</label>" : "<label class='choice'>" + proxy.getId()
-                            + "</label>");
-                    nodesPresent = true;
-                } else {
-                    LOGGER.warning("Node " + proxy.getId() + " does not support force restart.");
-                }
-            }
-
-            writer.write("</li>");
-            String defaultMsg = "<div style=\"padding-top: 20px\"><label class='choice'>No nodes are available to restart</label class='choice'></div>";
-            if (nodesPresent) {
-                defaultMsg = "<li class='buttons'><input type='hidden' name='form_id' value='restart_nodes' />"
-                        + "<input id='saveForm' class='button_text' type='submit' name='submit' value='Restart'/>"
-                        + "<input id='saveForm' class='button_text' type='submit' name='submit' value='Force Restart'/></li>";
-            }
-            writer.write(defaultMsg);
-            writer.write("</ul>");
-            writer.write("</form>");
-            ServletHelper.displayFooter(writer);
-            writer.write("</body>");
-            writer.write("</html>");
+            List<ProxyInfo> proxies = getProxyInfo();
+            final String json = new GsonBuilder().serializeNulls().create().toJson(proxies);
+            final String jsonUtf8 = new String(json.getBytes(), "UTF-8");
+            String template = IOUtils.toString(this.getClass().getResourceAsStream(RESOURCE_PAGE_FILE), "UTF-8");
+            writer.write(String.format(template, jsonUtf8, GridForceRestartDelegateServlet.class.getSimpleName()));
```

Comment:
I think we should create a `ServletHelper#respondAsHtmlUsingJsonAndTemplateWithHttpStatus(HttpServletResponse resp, Object toJson, String resourcePageTemplate, int statusCode, String ... additionalStrings)` to use here and in other places.